core/core-configuration-layer.el

```
@@ -2800,7 +2806,12 @@ ARGS: format string arguments."
 
 (defun configuration-layer/load-file (file &optional noerror)
   "Load file silently except if in debug mode."
-  (load file noerror (not init-file-debug)))
+  (condition-case-unless-debug err
+      (load file noerror (not init-file-debug))
+    ('error
+     (configuration-layer//error
+      "An error occurred while loading %S (error: %s)"
+      file err))))
```

I'm not convinced this change is a good idea.  `configuration-layer/load-file` is used in several places that IMO should simply propagate any errors loading the file.  For example in `configuration-layer/rollback` and `configuration-layer/load-auto-layer-file` (and maybe `configuration-layer/make-layer`).

It is also unfortunately a "public" function based on naming convention, so one more reason to maybe change the callsites rather than the function itself (although I consider this a weaker reason).

---

> configuration-layer/load-file is used in several places that IMO should simply propagate any errors loading the file.

In `configuration-layer/rollback` we should definitely propagate the error: State from a previous invocation could be left in `update-packages-alist`, which would execute those previous rollbacks again.

In `configuration-layer/load-auto-layer-file` and `configuration-layer/make-layer` I don't immediately see the necessity to propagate the errors. Do you think something bad could happen if we continue in case of errors? In particular, could failing to load a single file be realistically worse than not loading any other file after an error? I guess it could be, but I'm not convinced yet.